modules/packages/LinearAlgebra.chpl

```diff
@@ -1331,6 +1331,100 @@ proc solve (A: [?Adom] ?eltType, b: [?bdom] eltType) {
   return x;
 }
 
+/* Compute least-squares solution to ``A * x = b``.
+   Compute a vector ``x`` such that the 2-norm ``|b - A x|`` is minimized.
+
+   ``cond`` is the cut-off threshold such that singular values will be
+   considered 0.0. If ``cond < 0.0`` (defaults to ``-1.0``), the treshold will
+   be set to ``max((...A.shape)) * epsilon``, where ``epsilon`` is the machine
+   precision for ``A.eltType``.
+
+   Returns a tuple of ``(x, residues, rank, s)``, where:
+
+  - ``x`` is the the least-squares solution with shape of ``b``
+  - ``residues`` is the square of the 2-norm for each column in ``b - a x`` if ``M > N`` and ``A.rank == n``
+    - ``residues`` is a scalar if ``b`` is 1-D, otherwise it is a 1-element array
+  - ``rank`` is the effective rank of ``A``
```

Comment:
Ugh.. I wish `rank` wasn't overloaded :(

Reply:
Indeed. Do you have any suggestions?

Reply:
This is probably the curse of the LinearAlgebra module in general. It is just something that we need to live with, I think.

I was mostly being grumpy as it took me few seconds to understand what you are trying to say there :(